tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/EmailRecoveryActionSetMethodContract.scala

```diff
@@ -539,208 +542,181 @@ trait EmailRecoveryActionSetMethodContract {
+         |			{
+         |				"create": {
+         |					"clientId1": {}
+         |				}
+         |			},
+         |			"c1"
+         |		]
+         |	]
+         |}""".stripMargin
+    }
+
+    def createRequestWithCriterionDeletedBefore(deletedBefore: String): String = {
+      s"""{
+         |	"using": [
+         |		"urn:ietf:params:jmap:core",
+         |		"com:linagora:params:jmap:messages:vault"
+         |	],
+         |	"methodCalls": [
+         |		[
+         |			"EmailRecoveryAction/set",
+         |			{
+         |				"create": {
+         |					"clientId1": {
+         |						"deletedBefore": "$deletedBefore"
+         |					}
+         |				}
+         |			},
+         |			"c1"
+         |		]
+         |	]
+         |}""".stripMargin
+    }
+
+    def createRequestWithCriterionDeletedAfter(deletedAfter: String): String = {
+      s"""{
+         |	"using": [
+         |		"urn:ietf:params:jmap:core",
+         |		"com:linagora:params:jmap:messages:vault"
+         |	],
+         |	"methodCalls": [
+         |		[
+         |			"EmailRecoveryAction/set",
+         |			{
+         |				"create": {
+         |					"clientId1": {
+         |						"deletedAfter": "$deletedAfter"
+         |					}
+         |				}
+         |			},
+         |			"c1"
+         |		]
+         |	]
+         |}""".stripMargin
+    }
+
+    def createDeletedMail(server: GuiceJamesServer, deletionDate: ZonedDateTime) = {
+
       val mailboxId: MailboxId = server.getProbe(classOf[MailboxProbeImpl])
         .createMailbox(MailboxPath.inbox(BOB))
 
       val deletedMessage: DeletedMessage = templateDeletedMessage(
         messageId = randomMessageId,
-        mailboxId = mailboxId)
+        mailboxId = mailboxId,
+        deletionDate = deletionDate)
 
       server.getProbe(classOf[DeletedMessageVaultProbe])
-        .append(deletedMessage,
-          new ByteArrayInputStream(DELETED_MESSAGE_CONTENT))
-
-      assertThat(listAllMessageResult(server, BOB)).hasSize(0)
-
-      val deletedBefore = deletedMessage.getDeletionDate.plusHours(1)
-        .format(TIME_FORMATTER)
+        .append(deletedMessage, new ByteArrayInputStream(DELETED_MESSAGE_CONTENT))
+    }
 
-      val taskId: String = `given`
-        .body(
-          s"""{
-             |	"using": [
-             |		"urn:ietf:params:jmap:core",
-             |		"com:linagora:params:jmap:messages:vault"
-             |	],
-             |	"methodCalls": [
-             |		[
-             |			"EmailRecoveryAction/set",
-             |			{
-             |				"create": {
-             |					"clientId1": {
-             |						"deletedBefore": "$deletedBefore"
-             |					}
-             |				}
-             |			},
-             |			"c1"
-             |		]
-             |	]
-             |}""".stripMargin)
-      .when
+    private def sendRequestAndGetResponse(request: String): String = {
+      `given`
+        .body(request)
+        .when
         .post
-      .`then`
+        .`then`
```

Comment:
Please preserve the original ident which was intentionnal

(highlight restassured statement structure given-when-then...)